src/components/Drawer.tsx

```diff
@@ -0,0 +1,148 @@
+import { type CSSProperties, type PropsWithChildren, useEffect } from "react";
+import { createPortal } from "react-dom";
+import "./Drawer.css";
+
+export interface DrawerProps {
+  isOpen: boolean;
+  onClose: () => void;
+  anchor?: "left" | "top" | "right" | "bottom";
+  isBackdrop?: boolean;
+  width?: CSSProperties["width"];
+  height?: CSSProperties["height"];
+  zIndex?: CSSProperties["zIndex"];
+}
+
+const Drawer = (props: PropsWithChildren<DrawerProps>) => {
+  const {
+    anchor = "left",
+    isOpen,
+    isBackdrop = true,
+    onClose,
+    width,
+    height,
+    zIndex = 1,
+    children,
+  } = props;
+
+  useEffect(() => {
+    const handleKeydown = (e: KeyboardEvent) => {
+      if (e.key !== "Escape") {
+        return;
+      }
+
+      onClose();
```

Comment:
```suggestion
    const handleKeydown = (e: KeyboardEvent) => {
      if (e.key === "Escape") {
        onClose;
      }
```

이렇게 써도 되지 않았을까요?? 직관적으로요!

Reply:
오 그렇네요 고쳐둘게요!!